Automattic/wp-calypso
# Update core global styles panel events for new UI

#### Changes proposed in this Pull Request

Updates global styles panel events to fit with the newly updated interface.  Previously, we used 1 event to track both open/close of the panel as well as the selected tab context.  Now that tabs have been removed it makes sense to decouple this into two events.  1 for toggling the panel open/closed states, and another for selection of a menu context (typography, colors, blocks, etc.).

follow-up TODOs - after 11.6 lands on production, we merge this PR, and release the new `wpcom-block-editor` build:
* update tracks to remove the `wpcom_block_editor_global_styles_tab_selected` and add our two new replacement events `wpcom_block_editor_global_styles_panel_toggle` and `wpcom_block_editor_global_styles_menu_selected`.
* open another PR to test and release the updated e2e tests that are added with a `.skip` here. 

#### Testing instructions



* Build the `wpcom-block-editor` app and sync it to your sandbox.  
* Follow steps to enable tracking debugging at PCYsg-nrf-p2
* Visit the site editor on a site running gutenberg edge (or Gutenberg v11.6.1-rc.1 or greater).
* Open the global styles panel and verify the `wpcom_block_editor_global_styles_panel_toggle` event fires with the property `open` as `true`.
* Close the global styles panel and verify this same event fires with the `open` property as `false`.
* Open the global styles panel, then open the block inspector panel, verify the above event also fires with the `open` property as `false`.
* Open the global styles panel and select one of the menu options (Typography, Colors, or Blocks).  Verify that for this action the `wpcom_block_editor_global_styles_menu_selected` event fires with the `menu` property corresponding to the menu selected ('typography', 'colors', or 'blocks').



Related to #
